pkg/network/ebpf/c/sock.h

```diff
@@ -152,9 +154,9 @@ static __always_inline __u64 offset_socket_sk() {
 }
 
 static __always_inline __u32 get_netns_from_sock(struct sock* sk) {
-    possible_net_t* skc_net = NULL;
+    void* skc_net = NULL;
```

Comment:
I'm guessing this is related to your other header changes. Can you explain why it is necessary? I don't love having to redefine everything from kernel header files.

Reply:
So the reason why I got rid of some of these includes is because of their transient dependencies which end up including asm macros with constraints that are not that are not supported by LLVM for the BPF target.  With ` <net/net_namespace.h>` specifically you get this:

```
In file included from ./pkg/network/ebpf/c/prebuilt/http.c:5:
In file included from ./pkg/network/ebpf/c/sock.h:4:
In file included from /usr/src/linux-headers-5.4.0-52-generic/include/net/net_namespace.h:10:
In file included from /usr/src/linux-headers-5.4.0-52-generic/include/linux/workqueue.h:9:
In file included from /usr/src/linux-headers-5.4.0-52-generic/include/linux/timer.h:5:
In file included from /usr/src/linux-headers-5.4.0-52-generic/include/linux/list.h:9:
In file included from /usr/src/linux-headers-5.4.0-52-generic/include/linux/kernel.h:12:
In file included from /usr/src/linux-headers-5.4.0-52-generic/include/linux/bitops.h:26:
In file included from /usr/src/linux-headers-5.4.0-52-generic/arch/x86/include/asm/bitops.h:387:
/usr/src/linux-headers-5.4.0-52-generic/arch/x86/include/asm/arch_hweight.h:20:7: error: invalid output constraint '=a' in asm
                         : "="REG_OUT (res)
```

Reply:
This link has some useful context: https://www.mail-archive.com/linux-kernel@vger.kernel.org/msg1546605.html